elementary/clients/s3/client.py

```diff
@@ -22,15 +22,19 @@ def __init__(self, config: Config):
     def create_client(cls, config: Config) -> Optional['S3Client']:
         return cls(config) if config.has_aws else None
 
-    def send_report(self, html_path: str) -> bool:
+    # html_path is the local path where the report generated at.
+    # bucket_file_path is where we want to upload the report to at the bucket - support folders!
+    def send_report(self, html_path: str, bucket_file_path: Optional[str] = None) -> bool:
         report_filename = path.basename(html_path)
+        bucket_report_path = bucket_file_path if bucket_file_path else report_filename
         try:
-            self.client.upload_file(html_path, self.config.s3_bucket_name, report_filename,
+            self.client.upload_file(html_path, self.config.s3_bucket_name, bucket_report_path,
                                     ExtraArgs={'ContentType': 'text/html'})
             logger.info('Uploaded report to S3.')
             if self.config.update_bucket_website:
                 self.client.put_bucket_website(
                     Bucket=self.config.s3_bucket_name,
+                    # We use report_filename because a path can not be an IndexDocument Suffix.
```

Comment:
Will the website work if we upload it to a path and here we use only the name itself?

Reply:
Yeah it creates an index that can be access via the bucket website (under the folder path)